woocommerce/woocommerce-gateway-stripe
# Fix 3DS Authorize and Capture processing with Deferred Intent

Fixes #2764

## Changes proposed in this Pull Request:

This PR fixes an issue using 3DS cards with Auth and Capture enabled using our split UPE with deferred intent implementation so far on the `add/deferred-intent` branch. 

If you have auth and capture enabled, and process a payment using a SCA/3DS enabled card, the orders notes on the order will be missing the charge ID, and the order will be set to on-hold before the 3DS verification has been completed. This PR fixes these two issues.

## Testing instructions

1. Enable UPE and auth/capture in the Stripe settings. 
   - **WooCommerce → Settings → Payments → Stripe → Settings**
   - Enable the _'Issue an authorization on checkout, and capture later'_ setting
2. Place any simple product in your cart and enter a 3DS payment card number. eg `4000002500003155`
3. Don't complete the auth. 
4. Go to admin and view the order. On `add/deferred-intent`, you'll notice two things:
   1. The order note has no charge ID
   2. The order has an on-hold status before you've passed the 3DS auth.

| Order status | Order notes |
|--------|--------|
| <img width="400" alt="Screenshot 2023-12-06 at 5 00 54 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/17efa9c6-9c5c-44f4-85d2-26ceab776342"> | <img width="200" alt="Screenshot 2023-12-06 at 4 57 44 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/05f35850-1052-4502-a78a-69acd0985a25"> | 

5. You can complete the auth, the rest of the process for auth and capture works as expected.
6. Repeat the steps above on this branch you'll notice that
   - Unlike on `add/deferred-intent`, while the checkout hangs with the auth modal being shown, the order won't update it's status. It will remain pending payment.
   - There will be an order note about the payment intent being created but no reference to a charge yet.
7. Complete authorization and review the order again. 
8. It will now have an on-hold status and the order note with a charge ID. 
9. Verify this change doesn't break any other flow. 

---


**Post merge**
